src/py/migraphx_py.cpp

```diff
@@ -239,33 +247,54 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
     py::class_<migraphx::shape> shape_cls(m, "shape");
     shape_cls
         .def(py::init([](py::kwargs kwargs) {
-            auto v    = migraphx::to_value(kwargs);
-            auto t    = migraphx::shape::parse_type(v.get("type", "float"));
-            auto lens = v.get<std::size_t>("lens", {1});
-            if(v.contains("strides"))
-                return migraphx::shape(t, lens, v.at("strides").to_vector<std::size_t>());
+            auto v = migraphx::to_value(kwargs);
+            auto t = migraphx::shape::parse_type(v.get("type", "float"));
+            if(v.contains("dyn_dims"))
+            {
+                auto dyn_dims =
+                    kwargs["dyn_dims"].cast<std::vector<migraphx::shape::dynamic_dimension>>();
```

Comment:
This should do `from_value<std::vector<migraphx::shape::dynamic_dimension>>(v)`.